openlibrary/plugins/upstream/models.py

```diff
@@ -691,48 +698,118 @@ def get_sorted_editions(
         ebooks_only: bool = False,
         limit: int | None = None,
         keys: list[str] | None = None,
+        use_solr: bool = False,
     ) -> list[Edition]:
-        """
-        Get this work's editions sorted by publication year
-        :param list[str] keys: ensure keys included in fetched editions
-        """
-        db_query = {"type": "/type/edition", "works": self.key}
-        db_query['limit'] = limit or 10000  # type: ignore[assignment]
-
-        edition_keys = []
-        if ebooks_only:
-            if self._solr_data:
-                from openlibrary.book_providers import get_book_providers
-
-                # Always use solr data whether it's up to date or not
-                # to determine which providers this book has
-                # We only make additional queries when a
-                # trusted book provider identifier is present
-                for provider in get_book_providers(self._solr_data):
-                    query = {**db_query, **provider.editions_query}
-                    edition_keys += web.ctx.site.things(query)
-            else:
-                db_query["ocaid~"] = "*"
+        def get_solr_editions():
+            work = self._solr_data
+            edition_keys = work.get('edition_key') or []
+            edition_queries = [f'key:"/books/{key}"' for key in edition_keys]
+
+            ed_results = run_solr_query(
+                EditionSearchScheme(),
+                {'q': ' OR '.join(edition_queries)},
+                rows=limit,
+                fields=[
+                    'ia',
+                    'key',
+                    'title',
+                    'subtitle',
+                    'language',
+                    'isbn',
+                    'publish_date',
+                    'publisher',
+                ],
+                user_query=False,
+            )
 
-        if not edition_keys:
-            solr_is_up_to_date = (
-                self._solr_data
-                and self._solr_data.get('edition_key')
-                and len(self._solr_data.get('edition_key')) == self.edition_count
+            _editions = []
+            if ed_results.docs:
+                for solr_edition in ed_results.docs:
+                    ed = web.storify(solr_edition)
+                    if ed.get('ia', []):
+                        ed.ocaid = ed.ia[0]
```

Comment:
```suggestion
                    if ed.get('ia', ''):
                        ed.ocaid = ed.ia
```
`ia` is a string, not a list.  This is causing the wrong identifier to be used when decorating the edition with availability.